Phenotypes.pm

```diff
@@ -128,9 +130,12 @@ sub new {
 
   #DEFAULTS are not refreshed automatically by multiple REST calls unless forced
   my $refresh = 0;
-  $refresh = 1 if (exists $DEFAULTS{species} && $DEFAULTS{species} ne $self->{config}{species});
+  $refresh = 1 if (exists $CONFIG{species} && $CONFIG{species} ne $self->{config}{species});
+
+  unless($CONFIG{file} && !$refresh) {
+
+    die("ERROR: Unable to generate GFF file in offline mode\n") if $self->{config}->{offline};
```

Comment:
Do we need to add the following line as well here ?
```
 die("ERROR: Not allowed to generate GFF file in rest mode\n") if $config->{rest};
```

Reply:
Yes, otherwise there will be an undefined $assembly concatenation error if used without no file input + cmd line + offline.  After the undef code will exit in the next check within generate_phenotype_gff. This line is to prevent the undef concatenation errors.